tests/cephadm/test_client.py

```diff
@@ -0,0 +1,149 @@
+from ceph.utils import get_node_by_id
+
+log = logging.getLogger(__name__)
+
+
+def add(cls, config: Dict) -> None:
+    """
+    configure client using the provided configuration.
+
+    Args:
+        cls: cephadm object
+        config: Key/value pairs provided by the test case to create the client.
+
+    Example
+        config:
+            command: add
+            id: client.1                    # client Id
+            node: "node8"                   # client node
+            install_packages:
+              - ceph_common                 # install ceph common packages
+            copy_admin_keyring: true|false  # copy admin keyring
+            caps:                           # authorize client capabilities
+              - "mon 'allow r'"
+              - "osd 'allow rw pool=liverpool'"
+    """
+    id_ = config["id"]
+    node = get_node_by_id(cls.cluster, config["node"])
+
+    def put_file(client, file_name, content, file_mode, sudo=True):
+        file_ = client.remote_file(sudo=sudo, file_name=file_name, file_mode=file_mode)
+        file_.write(content)
+        file_.flush()
+
+    # Create client
+    cmd = ["ceph", "auth", "get-or-create", f"{id_}"]
+    [cmd.append(f"{k} '{v}'") for k, v in config.get("caps").items()]
+    cnt_key, err = cls.shell(args=cmd)
+
+    # Copy the keyring to client
+    node.exec_command(sudo=True, cmd="mkdir -p /etc/ceph")
+    put_file(node, f"/etc/ceph/ceph.{id_}.keyring", cnt_key, "w")
+
+    # Get minimal ceph.conf
+    ceph_conf, err = cls.shell(args=["ceph", "config", "generate-minimal-conf"])
+
+    # Copy the ceph.conf to client
+    put_file(node, "/etc/ceph/ceph.conf", ceph_conf, "w")
+
+    # Copy admin keyring to client node
+    admin_keyring, _ = cls.shell(args=["ceph", "auth", "get", "client.admin"])
+    put_file(node, "/etc/ceph/ceph.client.admin.keyring", admin_keyring, "w")
+
+    # Install ceph-common
+    if config.get("install_packages"):
+        for pkg in config.get("install_packages"):
+            node.exec_command(cmd=f"yum install -y --nogpgcheck {pkg}", sudo=True)
+
+    out, _ = node.exec_command(cmd="ls -ltrh /etc/ceph/", sudo=True)
+    log.info(out.read().decode().strip())
+
+
+def remove(cls, config: Dict) -> None:
+    """
+    configure client using the provided configuration.
+
+    Args:
+        cls: cephadm object
+        config: Key/value pairs provided by the test case to create the client.
+
+    Example
+        config:
+            command: remove
+            id: client.0                # client Id
+            node: "node8"               # client node
+            remove_packages:
+                - ceph-common           # Remove ceph common packages
+            remove_admin_keyring: true  # Copy admin keyring to node
+    """
+    node = get_node_by_id(cls.cluster, config["node"])
+    id_ = config["id"]
+
+    cls.shell(
+        args=["ceph", "auth", "del", id_],
+    )
+
+    if config.get("remove_admin_keyring"):
```

Comment:
If we chose not to have `admin_keyring` flag in the config, we may want to change this statement.

Reply:
we needed admin key-ring as mentioned in the above comment.